sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -590,6 +590,10 @@ class InMemoryCatalog(
     catalog(db).functions.remove(funcName)
   }
 
+  override protected def doAlterFunction(db: String, func: CatalogFunction): Unit = synchronized {
+    // do nothing
```

Comment:
I implement this in my first commit, but gatorsmile suggest me that there is no need to impl this, So i'm not sure whether should we impl this in `ImemoryCatalog` or not.